### ISSUE TITLE
Warning "Mixin base type should be the same as parent base type" on parent which is implicit object

### ISSUE THREAD

**Issue by honzajavorek** [apiaryio/drafter#243]
I get warning from Atom's API Blueprint Linter

> Mixin base type should be the same as parent base type, objects should contain object mixins, arrays should contain array mixins

when specifying mixin on object without explicit type:

<img width="979" alt="screen shot 2016-02-25 at 10 40 29" src="https://cloud.githubusercontent.com/assets/283441/13315726/db6d970a-dbac-11e5-8e3c-769c3a8a4797.png">

```apib
FORMAT: 1A

# Building Supply Store API

## Brick Type [/bricks/{id}]

+ Parameters
    + id: XYZ42 - ID of a brick type.
+ Attributes
    + id
    + name (string, required)
    + colors: red, brown (array)
    + Include DimensionsMixin

## Data Structures

### DimensionsMixin (object)

+ dimensions (array)
    + 20, 30, 40 (array[number])
```

Compare with:

<img width="463" alt="screen shot 2016-02-25 at 10 40 42" src="https://cloud.githubusercontent.com/assets/283441/13315730/e1f66fa2-dbac-11e5-9d31-22ee263ccaaf.png">

```apib
FORMAT: 1A

# Building Supply Store API

## Brick Type [/bricks/{id}]

+ Parameters
    + id: XYZ42 - ID of a brick type.
+ Attributes (object)
    + id
    + name (string, required)
    + colors: red, brown (array)
    + Include DimensionsMixin

## Data Structures

### DimensionsMixin (object)

+ dimensions (array)
    + 20, 30, 40 (array[number])
```

**Comment by honzajavorek** [comment]
As tested in https://github.com/apiaryio/protagonist-include-issue, this issue occurs also when doing following:

```apib
+ Attributes
    + id
    + name (string, required)
    + colors: red, brown (array)
    + Include DimensionsMixin

...

## Data Structures

### DimensionsMixin (object)

+ dimensions
    + height: 20 (number)
    + width: 30 (number)
    + length: 40 (number)
```

I get the same warning, although there are _no arrays_ in my mixin!

**Comment by pksunkara** [comment]
This might be because of another bug. https://github.com/apiaryio/snowcrash/issues/362.

Also related to #255 #254

**Comment by honzajavorek** [comment]
@pksunkara Yes, I think @kylef did some research about this last Thursday and came to conclusion that it's still the same problem and resolving it would probably close many issues on Drafter/Snowcrash. It seemed to me that the https://github.com/apiaryio/snowcrash/issues/362 is probably the oldest record for the issue, the rest are sort of duplicates or very close issues.

**Comment by pksunkara** [comment]
Fixed by #280
